### DISCUSSION TITLE
How to log errors/exceptions

### DISCUSSION THREAD

**Discussion by DeuSCaNoN** [uNetworking/uWebSockets#1462] [Q&A]
So I'm using this library and it's pretty simple so I deployed my server on AWS. The issue is that on AWS my application will crash and I can't reproduce it locally.

I tried to put my entire main body inside a try catch but that doesn't seem to help either.

Does anyone have any idea how I can log the kinds of errors that uWebsockets will produce?

```
int main()
{
	try 
	{
		... // all the code to start up my server
	}
	catch (std::exception& e) // Does uWebsocket even throw std::exceptions or something else?
	{
		using namespace std;
		freopen("output.txt", "w", stdout);
		cout << "exception: " << e.what() << endl;
	}

	return 0;
}
```

**Comment by ghost** [comment]
Sounds like you want a debugger like gdb or compile WITH_ASAN

**Reply by DeuSCaNoN** [reply]
Ideally yes, i want to debug it as it happens, but like i said I can't reproduce it locally as I don't know the situations that caused it to happen when it was open to the open web.

I'm guessing WITH_ASAN is a logging preprocessor flag? Is there documentation on how to use it?

**Reply by ghost** [reply]
https://github.com/uNetworking/uWebSockets/blob/c2f29c86c718e364b5586d103ad592aa396d7604/Makefile#L54

If you build with "WITH_ASAN=1 make" you get pretty much a stand-alone debugger attached to your build.

**Reply by DeuSCaNoN** [reply]
ok thanks a lot, hopefully, this works in my environment since I build with Visual Studio, but I didn't know about these debugger-like addons to builds so if not I have an avenue to chase down.

**Reply by ghost** [reply]
Generally speaking, Windows is not a platform you want to write C++ on. It's been horrible since at least 1996. Use WSL2 and Ubuntu and save yourself from immense mental suffering.

**Reply by DeuSCaNoN** [reply]
lol, thanks for the advice. I think at this point it's just laziness that I don't want to relearn the workflow to develop on Linux and just continue my masochistic suffering in windows. I'll take that to heart though and maybe one day swap over to Linux.
